test/api/capi/test_capi_appender.cpp

```diff
@@ -677,6 +677,173 @@ TEST_CASE("Test append timestamp in C API", "[capi]") {
 	REQUIRE(result->Fetch<string>(0, 0) == "2022-04-09 15:56:37.544");
 }
 
+TEST_CASE("Test append timestamp_s in C API", "[capi]") {
+	CAPITester tester;
+	duckdb::unique_ptr<CAPIResult> result;
+	REQUIRE(tester.OpenDatabase(nullptr));
+
+	tester.Query("CREATE TABLE test (t timestamp_s)");
+	duckdb_appender appender;
+
+	auto status = duckdb_appender_create_ext(tester.connection, nullptr, nullptr, "test", &appender);
+	REQUIRE(status == DuckDBSuccess);
+	REQUIRE(duckdb_appender_error(appender) == nullptr);
+
+	duckdb_date_struct date_struct_in;
+	date_struct_in.year = 1992;
+	date_struct_in.month = 9;
+	date_struct_in.day = 3;
+
+	duckdb_time_struct time_struct_in;
+	time_struct_in.hour = 12;
+	time_struct_in.min = 22;
+	time_struct_in.sec = 33;
+	time_struct_in.micros = 0;
+
+	duckdb_timestamp_struct ts_struct_in;
+	ts_struct_in.date = date_struct_in;
+	ts_struct_in.time = time_struct_in;
+
+	duckdb_timestamp_s ts_s_in = duckdb_to_timestamp_s(ts_struct_in);
+	REQUIRE(ts_s_in.seconds == 715522953);
+
+	REQUIRE(duckdb_appender_begin_row(appender) == DuckDBSuccess);
+	REQUIRE(duckdb_append_timestamp_s(appender, ts_s_in) == DuckDBSuccess);
+	REQUIRE(duckdb_appender_end_row(appender) == DuckDBSuccess);
+
+	REQUIRE(duckdb_appender_flush(appender) == DuckDBSuccess);
+	REQUIRE(duckdb_appender_close(appender) == DuckDBSuccess);
+	REQUIRE(duckdb_appender_destroy(&appender) == DuckDBSuccess);
+
+	result = tester.Query("SELECT * FROM test");
+	REQUIRE_NO_FAIL(*result);
+	auto chunk = result->FetchChunk(0);
+	auto data = reinterpret_cast<duckdb_timestamp_s *>(chunk->GetData(0));
+	auto ts_s_out = data[0];
```

Comment:
`auto ts_s_out = result->Fetch<duckdb_timestamp_s>(0, 0);` should do the trick here and in the other tests.

Reply:
Unfortunately it doesn't. This gave me quite a headache, actually.

I started out using that pattern, following how it's done elsewhere, but I found the resulting values were off. For example, for TIMESTAMP_S, I'd get back a value in microseconds (e.g. 1,000,000 times the value I expected).

I tracked this down to how the various overloads of CAPIResult::Fetch work. They all use either one of the (deprecated) `duckdb_value_xxx` functions, or the (deprecated) `duckdb_column_data` function. Since there aren't `duckdb_value_xxx` functions for the non-standard timestamp types, and I don't want to add new functions to a deprecated set, I tried to use `duckdb_column_data`. But this uses `DeprecatedMaterializeResult`, which, while materializing, does some casting, including converting the non-standard timestamp types to microsecond timestamps. Hence the unexpected values I was seeing.

I didn't want to invest in updating this deprecated path, and perhaps more importantly it would result in a behavior change visible through the C API (and that seems like a no-go). So I opted to fetch the data directly, rather than using the tester to fetch it.

Reply:
All right, I agree that we should not add more deprecated functions or change any deprecated paths. Let's keep it like this. :)